### DISCUSSION TITLE
Pickle/Unpickle query behaving differently depending on column name

### DISCUSSION THREAD

**Discussion by rad-pat** [sqlalchemy/sqlalchemy#6248] [Usage Questions]
I have a query which is evaluated in a sandboxed environment which is passed back for execution and I have been using pickle/unpickle for this. 

With v1.4, I am getting slightly different behaviour depending on if a column has a space in the column name or not. Any ideas why?

By example, 
if COL2 has a space, I get a `psycopg2.errors.SyntaxError: syntax error at or near "["` 
if it has no space, it works fine
```py
import sqlalchemy
from sqlalchemy import Table, Column, String, Integer, and_
import pickle


#COL2 = 'column2'
COL2 = 'column 2'

metadata = sqlalchemy.MetaData()

user_table = Table(
    'user', metadata,
    Column('id', Integer, primary_key=True),
    Column(COL2, String(50)),
)


engine = sqlalchemy.create_engine('postgresql://.....')

connection = engine.connect()

sel = sqlalchemy.select(
    user_table.columns
).where(
    and_(
        user_table.columns['id'] == 1,
        user_table.columns[COL2].in_(['1', '2', '3'])
    )
)


p = pickle.dumps(sel)

# clean meta data to imitate loading unpickling on the other end
user_table = None
metadata = None

sel = pickle.loads(p)

engine.execute(sel)
```

Thanks for any help.

**Comment by zzzeek** [comment]
assuming this worked in 1.3 this would be a regression in SQLAlchemy.

**Reply by rad-pat** [reply]
It does work ok with 1.3.24

**Reply by zzzeek** [reply]
sure is a reg

**Reply by zzzeek** [reply]
https://github.com/sqlalchemy/sqlalchemy/issues/6249 this will be fixed in SQLAlchemy 1.4.8

**Reply by rad-pat** [reply]
👍
